### DISCUSSION TITLE
运行inference_hf.py遇到的错误不知道怎么解决

### DISCUSSION THREAD

**Discussion by dqy1111** [zai-org/ChatGLM3#1077] [Q&A] [answered]
root@dsw-337780-78cf4f7476-mccgv:/mnt/workspace/ChatGLM3/finetune_demo# python inference_hf.py output/checkpoint-3000/ --prompt "类型#裙版型#显瘦材质#网纱风格#性感裙型#百褶 裙下摆#压褶裙长#连衣裙* 裙衣门襟#拉链裙衣门襟#套头裙款式#拼接裙款式#拉链裙款式#木耳边裙款式#抽褶裙款式#不规则"
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /mnt/workspace/ChatGLM3/finetune_demo/inference_hf.py:50 in main                                 │
│                                                                                                  │
│   47 │   │   model_dir: Annotated[str, typer.Argument(help='')],                                 │
│   48 │   │   prompt: Annotated[str, typer.Option(help='')],                                      │
│   49 ):                                                                                          │
│ ❱ 50 │   model, tokenizer = load_model_and_tokenizer(model_dir)                                  │
│   51 │   response, _ = model.chat(tokenizer, prompt)                                             │
│   52 │   print(response)                                                                         │
│   53                                                                                             │
│                                                                                                  │
│ /mnt/workspace/ChatGLM3/finetune_demo/inference_hf.py:30 in load_model_and_tokenizer             │
│                                                                                                  │
│   27 def load_model_and_tokenizer(model_dir: Union[str, Path]) -> tuple[ModelType, TokenizerT    │
│   28 │   model_dir = _resolve_path(model_dir)                                                    │
│   29 │   if (model_dir / 'adapter_config.json').exists():                                        │
│ ❱ 30 │   │   model = AutoPeftModelForCausalLM.from_pretrained(                                   │
│   31 │   │   │   model_dir, trust_remote_code=True, device_map='auto'                            │
│   32 │   │   )                                                                                   │
│   33 │   │   tokenizer_dir = model.peft_config['default'].base_model_name_or_path                │
│                                                                                                  │
│ /opt/conda/lib/python3.10/site-packages/peft/auto.py:69 in from_pretrained                       │
│                                                                                                  │
│    66 │   │   are passed along to `PeftConfig` that automatically takes care of filtering the    │
│    67 │   │   the config object init.                                                            │
│    68 │   │   """                                                                                │
│ ❱  69 │   │   peft_config = PeftConfig.from_pretrained(pretrained_model_name_or_path, **kwargs   │
│    70 │   │   base_model_path = peft_config.base_model_name_or_path                              │
│    71 │   │                                                                                      │
│    72 │   │   task_type = getattr(peft_config, "task_type", None)                                │
│                                                                                                  │
│ /opt/conda/lib/python3.10/site-packages/peft/config.py:137 in from_pretrained                    │
│                                                                                                  │
│   134 │   │   │   config_cls = cls                                                               │
│   135 │   │                                                                                      │
│   136 │   │   kwargs = {**class_kwargs, **loaded_attributes}                                     │
│ ❱ 137 │   │   config = config_cls(**kwargs)                                                      │
│   138 │   │   return config                                                                      │
│   139 │                                                                                          │
│   140 │   @classmethod                                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: LoraConfig.__init__() got an unexpected keyword argument 'layer_replication'

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
更新到最新代码试试？
